docs/how-tos/scaling.rst

```diff
@@ -0,0 +1,133 @@
+.. _scaling:
+
+Scaling your deployment
+***********************
+
+By default, the `Aspects Tutor plugin`_ deploys single nodes for these services:
+
+* :ref:`quick-start-ralph` or :ref:`quick-start-vector`
+* Clickhouse: ephemeral single node cluster
+* Superset: shares Open edX's MySQL and Redis instances
+
+Most deployments will benefit greatly from scaling horizontally and/or vertically, especially when running Aspects. Aspects is fed by Open edX event data, which is triggered by user actions on the site, and Open edX can generate a lot of event data. The initial processing of this event data occurs in platform plugins, so scaling your LMS workers speeds this processing time. And the faster events can be processed, the quicker you will see them appear in Aspects' dashboards and charts.
+
+And all production deployments will need a `persistent Clickhouse cluster`_.
+
+Preparing the LMS workers
+=========================
+
+Before deploying Aspects, we recommend :ref:`event_bus`.
+
+Or, if you'd rather use celery, we recommend boosting the number of LMS workers to prepare the celery queue for the high-volume of Aspects tasks.
+
+These plugins can be used to configure and enable autoscaling for the LMS and CMS. See their READMEs for details:
+
+* `tutor-contrib-pod-autoscaling`_ : for single instance deployments
+* `tutor-contrib-grove`_ : for multiple instance deployments
+
+We also recommend configuring the LMS to use the high-priority celery queue for Aspects tasks (`platform-plugin-aspects event sinks`_, `event-routing-backends xAPI tasks`_). This leaves the low-priority queue clear for other LMS tasks.
+
+.. code-block:: yaml
+
+  TBD -- implement configurable queue for Aspects tasks
```

Comment:
From [comment](https://github.com/openedx/openedx-aspects/issues/78#issuecomment-1910413023):

> It would be better to move the aspects-related tasks to the high queue, so it will perform better and doesn't block other LMS tasks.

@Ian2012 @bmtcril  IIRR, to do this we need to do this for both the [platform-plugin-aspects event sinks](https://github.com/openedx/platform-plugin-aspects/blob/main/platform_plugin_aspects/tasks.py) and [event-routing-backends tasks](https://github.com/openedx/event-routing-backends/blob/master/event_routing_backends/tasks.py) tasks, correct?

* add an app setting to configure the default celery queue. Default to `HIGH_PRIORITY_QUEUE`.
* pass this queue setting as to the `queue` parameter wherever the tasks are run.

Reply:
And then create a new consumer for both services (lms, cms) named `aspects-{{service}}-consumer` which are configured as any other lms-cms worker but that will only read that queue

Reply:
Oo cool.. will await the results of your experiment, thank you!

Reply:
@bmtcril @Ian2012 Should we enable batching by default, if it helps performance this much?

Reply:
I didn't want to modify the default behaviorin ERB and I think we can enable it with the tutor plugin